### DISCUSSION TITLE
Compilation Error in 05-local-execution.ipynb

### DISCUSSION THREAD

**Discussion by Zaia-Pieterjan** [aurelio-labs/semantic-router#163] [Q&A]
**Issue Description:**

I'm encountering an error while trying to run `05-local-execution.ipynb`. Specifically, when I reach the following code snippet:

```python
out = rl("what's the time in New York right now?")
print(out)
get_time(**out.function_call)
```

I receive the following error message:
ERROR semantic_router.utils.logger LLM error: No output generated
<img width="1149" alt="Screenshot 2024-02-20 at 18 06 31" src="https://github.com/aurelio-labs/semantic-router/assets/160490340/b7238910-2254-4578-a386-e1e93946bc21">
<img width="1149" alt="Screenshot 2024-02-20 at 18 06 41" src="https://github.com/aurelio-labs/semantic-router/assets/160490340/a1d9097b-ad7f-49d5-82fd-428a93dd3dca">

**Comment by octocat20** [comment]
That is not a compile error. The local LLM returned empty content while extracting function args.

Dynamic routes need llama.cpp to spit out the JSON for get_time. If content is empty, LlamaCppLLM raises "No output generated". The route can still match and arg extraction still fails.

Quick checks:
- Make sure the GGUF actually chats on its own before wiring SemanticRouter
- Use get_time(**out.function_call[0]). function_call is a list now. Old **out.function_call breaks even when the model works
- If function_call is None/empty, do not unpack it. Extraction failed

Schema wants IANA zones like America/New_York. Weaker local models sometimes blank on that. If it keeps happening, fix the GGUF / llama.cpp setup first or try a stronger quant.
